Sources/Plugins/SRTMPlugin/OASRTMPlugin.mm

```diff
@@ -88,11 +88,14 @@ - (void) onProfileSettingSet:(NSNotification *)notification
 {
     OACommonPreference *obj = notification.object;
     if (obj == _enable3DMaps)
-    {
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [OARootViewController.instance.mapPanel.mapViewController recreateHeightmapProvider];
-        });
-    }
+        [self updateLayers];
+}
+
+- (void)updateLayers
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [OARootViewController.instance.mapPanel.mapViewController recreateHeightmapProvider];
```

Comment:
Looks like updateLayers is called from updateCurrentMapSource like recreateHeightmapProvider. So it might be wrong fix.